protocols/mdns/src/behaviour/iface.rs

```diff
@@ -319,10 +316,9 @@ where
                 Poll::Ready(Ok(Err(err))) => {
                     tracing::debug!("Parsing mdns packet failed: {:?}", err);
                 }
-                Poll::Ready(Ok(Ok(None))) | Poll::Pending => {}
+                Poll::Ready(Ok(Ok(None))) => {}
+                Poll::Pending => return Poll::Pending,
```

Comment:
For consistency with other parts of the codebase, we tend to have the `return Poll::Pending` at the end of the loop and explicitly `continue` in all other branches. I find that easier to audit although it doesn't prevent us from mistakes as we can this in this PR :)

Reply:
Ok will do :wink: Indeed we have to be really careful that no unwanted `Poll::Ready` hits the `return Poll::Pending` when we don't want to.